### PR TITLE
gce.sh: Be strict about fullnode count w/o --allow-boot-failures

### PR DESCRIPTION
**Pull request by t-nelson**
#### Problem

`net/gce.sh` happily allocates a testnet with fewer than the requested number of fullnodes so long as more than zero showed up, even without `--allow-boot-failures` (formerly `-f`).

#### Summary of Changes

Strictly check the the fullnode count unless `--allow-boot-failures` was passed

### CODE REVIEW

**Comment by garious** [comment]
Uh oh, the word "fullnode" is being added to new code? What inspired that?

**Comment by t-nelson** [comment]
Nope!  Reuse of in-place legacy terminology 🙂 .  I can reword the commit message if you like?

File: net/gce.sh [hunk 1]
```diff
@@ -487,10 +487,24 @@ EOF
   fi
 
   if [[ $additionalFullNodeCount -gt 0 ]]; then
-    for zone in "${zones[@]}"; do
+    num_zones=${#zones[@]}
```
Comment (mvines) [inline net/gce.sh]: numZones

**Comment by garious** [comment]
I blame me for this. I see `fullnode` is still defined in the book.  Killing that here: https://github.com/solana-labs/solana/pull/6324

**Comment by t-nelson** [comment]
@garious ok if this goes in in the meantime? 🙏 

@mvines nits in https://github.com/solana-labs/solana/pull/6321/commits/8a15a622b4b40f60cac6184cd1ffe8e41f74a5c7

**Comment by mvines** [comment]
Of course.  Land it

**Comment by garious** [comment]
If I feel the need to block a PR, I'll use GitHub's "Request Changes" mechanism. Not going to do that over a variable name!

**Comment by t-nelson** [comment]
@danpaul000 already in. https://github.com/solana-labs/solana/pull/6318

**Pull request merged**